docs/references/message-execution-properties.mdx

```diff
@@ -14,26 +14,26 @@ The [community conversation on security best practices](https://www.youtube.com/
 
 To interact with a canister's methods, there are two primary types of **calls** that can be used: [update](/references/ic-interface-spec.md#http-call) calls and [query](/docs/current/references/ic-interface-spec#http-query) calls.
 
-If the Rust CDK is used, these are usually annotated with `#[update]` or `#[query]`, respectively. In Motoko, updates are declared as `public func`, and queries us the dedicated keyword: `public query func`. ICP also supports additional [entry points](/references/ic-interface-spec.md#entry-points) such as heartbeats, timers, initialization or upgrade hooks.
+The call must be addressed to an entry point of a canister, which is called a **method**. If the Rust CDK is used, these are usually annotated with `#[update]` or `#[query]`, respectively. In Motoko, updates are declared as `public func`, and queries use the dedicated keyword: `public query func`. ICP also supports additional [entry points](/references/ic-interface-spec.md#entry-points) such as heartbeats, timers, initialization or upgrade hooks.
 
-A **message** is a set of consecutive instructions that a subnet executes for a canister. The code execution for any such entry point can be split into several messages if inter-canister calls are made. The following properties are essential:
+A **message execution** is a set of consecutive instructions that a subnet executes when a canister's entry point is invoked. The code execution for any such entry point can be split into several message executions if inter-canister calls are made. The following properties are essential:
```

Comment:
I would try to introduce the "atomic" keyword somewhere in the first sentence.

A message execution can also happen for a response to a heartbeat or timer, so not only when invoking an endpoint.

In the second sentence, I would use direct language "if it makes inter-canister calls", to make it clear(er) that it's waiting for the downstream call responses that cause the split. Maybe even say "downstream calls".